### ISSUE TITLE
async log is slower than others

### ISSUE THREAD

**Issue by WinterChen** [gabime/spdlog#859]
Below are some benchmarks (https://github.com/gabime/spdlog/blob/v1.x/bench/bench.cpp) done in my centos computer, which has 64 processor(Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz)。
the async log is slower than others, it is different of the official result, why?

$ ./bench          
*******************************************************************************
Single thread, 1,000,000 iterations
*******************************************************************************
basic_st...             Elapsed: 0.347748       2,875,647/sec
rotating_st...          Elapsed: 0.361321       2,767,620/sec
daily_st...             Elapsed: 0.368148       2,716,295/sec
null_st...              Elapsed: 0.192847       5,185,455/sec

*******************************************************************************
10 threads sharing same logger, 1,000,000 iterations
*******************************************************************************
basic_mt...             Elapsed: 0.896685       1,115,218/sec
rotating_mt...          Elapsed: 0.852652       1,172,811/sec
daily_mt...             Elapsed: 0.928587       1,076,904/sec
null_mt...              Elapsed: 0.27646        3,617,163/sec

*******************************************************************************
async logging.. 10 threads sharing same logger, 1,000,000 iterations 
*******************************************************************************
async...                Elapsed: 1.03332        967,752/sec
async...                Elapsed: 1.10413        905,692/sec
async...                Elapsed: 1.01443        985,774/sec

**Comment by gabime** [comment]
Hard to tell.. Maybe related to the CPU differences? 
Which os and compiler are you using?

**Comment by WinterChen** [comment]
OS: centos 7
gcc: version 4.8.5 20150623
In spdlog, the async logger uses  multi producer-multi consumer blocking queue to store the messages form producer-threads. Every thread writes message, by invoking mpmc_blocking_queue::enqueue, it should get the mutex lock. I think the mutex competition between multi-threads is the reason for the slow throughput of writing log.

**Comment by gabime** [comment]
Right, but the question is why is it slower than the 10 threads tests. In the oficial bench it is about 60% faster.
